x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/bulk_actions/components/toolbar.tsx

```diff
@@ -0,0 +1,200 @@
+        return (
+          <EuiContextMenuItem
+            key={item.key}
+            data-test-subj={item['data-test-subj']}
+            disabled={isDisabled}
+            onClick={() => {
+              const selectedAlertIds = selectedIdsToTimelineItemMapper(alerts, rowSelection);
+              item.onClick(selectedAlertIds, isAllSelected);
+            }}
+          >
+            {isDisabled && item.disabledLabel ? item.disabledLabel : item.label}
+          </EuiContextMenuItem>
+        );
+      }),
+    [alerts, isAllSelected, items, rowSelection]
+  );
+
+  return bulkActionsItems;
+};
+
+const BulkActionsComponent: React.FC<BulkActionsProps> = ({ totalItems, items, alerts }) => {
+  const [{ rowSelection, isAllSelected }, updateSelectedRows] = useContext(BulkActionsContext);
+  const [isActionsPopoverOpen, setIsActionsPopoverOpen] = useState(false);
+  const [defaultNumberFormat] = useUiSetting$<string>(DEFAULT_NUMBER_FORMAT);
+  const [showClearSelection, setShowClearSelectiong] = useState(false);
+  const bulkActionItems = useBulkActionsToMenuItemMapper(items, alerts);
+
+  useEffect(() => {
+    setShowClearSelectiong(isAllSelected);
+  }, [isAllSelected]);
+
+  const selectedCount = rowSelection.size;
+
+  const formattedTotalCount = useMemo(
+    () => numeral(totalItems).format(defaultNumberFormat),
+    [defaultNumberFormat, totalItems]
+  );
+  const formattedSelectedEventsCount = useMemo(
+    () => numeral(selectedCount).format(defaultNumberFormat),
+    [defaultNumberFormat, selectedCount]
+  );
+
+  const toggleIsActionOpen = useCallback(() => {
+    setIsActionsPopoverOpen((currentIsOpen) => !currentIsOpen);
+  }, [setIsActionsPopoverOpen]);
+
+  const closeActionPopover = useCallback(() => {
+    setIsActionsPopoverOpen(false);
+  }, [setIsActionsPopoverOpen]);
+
+  const closeIfPopoverIsOpen = useCallback(() => {
+    if (isActionsPopoverOpen) {
+      setIsActionsPopoverOpen(false);
+    }
+  }, [isActionsPopoverOpen]);
+
+  const toggleSelectAll = useCallback(() => {
+    if (!showClearSelection) {
+      updateSelectedRows({ action: BulkActionsVerbs.selectAll });
+    } else {
+      updateSelectedRows({ action: BulkActionsVerbs.clear });
+    }
+  }, [showClearSelection, updateSelectedRows]);
+
+  const selectedAlertsText = useMemo(
+    () =>
+      showClearSelection
+        ? i18n.SELECTED_ALERTS(formattedTotalCount, totalItems)
+        : i18n.SELECTED_ALERTS(formattedSelectedEventsCount, selectedCount),
+    [
+      showClearSelection,
+      formattedTotalCount,
+      formattedSelectedEventsCount,
+      totalItems,
+      selectedCount,
+    ]
+  );
+
+  const selectClearAllAlertsText = useMemo(
+    () =>
+      showClearSelection
+        ? i18n.CLEAR_SELECTION
+        : i18n.SELECT_ALL_ALERTS(formattedTotalCount, totalItems),
+    [showClearSelection, formattedTotalCount, totalItems]
+  );
+
+  return (
+    <div
+      style={{ display: 'inline-block', position: 'relative' }}
```

Comment:
nit -> I will create a const to avoid the creation of this `div` at each rendering